narwhals/_pandas_like/series.py

```diff
@@ -433,9 +433,31 @@ def is_null(self) -> PandasLikeSeries:
         ser = self._native_series
         return self._from_native_series(ser.isna())
 
-    def fill_null(self, value: Any) -> PandasLikeSeries:
-        ser = self._native_series
-        return self._from_native_series(ser.fillna(value))
+    def fill_null(
+        self,
+        value: Any | None = None,
+        strategy: Literal["forward", "backward"] | None = None,
+        limit: int | None = None,
+    ) -> PandasLikeSeries:
+        if value is not None and strategy is not None:
+            msg = "cannot specify both `value` and `strategy`"
+            raise ValueError(msg)
+        if value is None and strategy is None:
+            msg = "must specify either a fill `value` or `strategy`"
+            raise ValueError(msg)
+        if strategy is not None and strategy not in {"forward", "backward"}:
+            msg = f"strategy not supported: {strategy}"
+            raise ValueError(msg)
```

Comment:
Commenting it here but it holds generally: I think we can move this validation block to narwhals Expr and Series directly. It would be easier to maintain and otherwise it would be possible to pass other strategy values to polars (e.g. `strategy="max"`).

Reply:
You're right! I've made the changes as suggested and moved the validation block to Expr and Series in Narwhal for easier maintenance. Thanks for the feedback!